sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -350,7 +367,8 @@ case class FileSourceScanExec(
         "PartitionFilters" -> seqToString(partitionFilters),
         "PushedFilters" -> seqToString(pushedDownFilters),
         "DataFilters" -> seqToString(dataFilters),
-        "Location" -> locationDesc)
+        "Location" -> locationDesc,
+        "ScanMode" -> scanMode.toString)
```

Comment:
Will replace `Batched` with `ScanMode`. Keep `Batched` here to make sure some existing unit tests to pass. If we are good with the overall approach of PR, I will make the change here and fix more unit tests.